app/controllers/admin/api/member_permissions_controller.rb

```diff
@@ -65,7 +65,10 @@ def user
   end
 
   def permission_params
-    params.require(:permissions).permit(allowed_service_ids: [], allowed_sections: [])
+    # both `:allowed_service_ids` and `allowed_service_ids: []` are required for service IDs
+    # because we want to allow empty value (`allowed_service_ids%5B%5D=%5B%5D`) and
+    # 'nil' value (`allowed_service_ids%5B%5D` or just `allowed_service_ids`)
+    params.require(:permissions).permit(:allowed_service_ids, allowed_service_ids: [], allowed_sections: [])
```

Comment:
Then why don't we do the same for `allowed_sections` and we have it like `params.require(:permissions).permit(:allowed_service_ids, allowed_service_ids: [], :allowed_sections, allowed_sections: [])` ?

Reply:
Considering this: 😄 
![image](https://user-images.githubusercontent.com/11318903/49524576-52156f80-f8ac-11e8-82e7-7003d42df625.png)

Reply:
@Martouta because we don't need to set `allowed_sections` to `nil`.
For allowed sections there just just two states:
- empty array `[]`
- an array of allowed values, e.g. `['partners', 'settings']`

For allowed service IDs, as explained in the PR description, we need three states, because each of them has its unique meaning:
- empty array `[]` - means "no services are allowed"
- an array of valid IDs, e.g. `[12,14,21]`
- a `nil` value (that will not be returned in the response), which means that "all current and future services are enabled". For `allowed_sections` we don't need such thing, as the list of sections is fixed and not dynamic as in case of services.

If you have another suggestion how to handle all these scenarios, please suggest.
I originally discussed this with @mikz too, and I thought there was an agreement :)

We could look into introducing some "special value", say, `"all"`, and make `allowed_service_ids[]=all` enable all services. But then it would make sense also reflect it in the response, so it would be messy and confusion.

Another option I was thinking of is adding a yet another field, say `all_service_ids_allowed`: `true`/`false`.
But then it would be a whole another story on how to resolve conflicts between this and `allowed_service_ids`, and changing the response answer would also cause many extra changes.

Reply:
Right, sorry, I didn't realise about this when thinking about the params 🤦‍♀️ 
Ok, then I think this is the simplest solution 😄